weather_data_analyzer.py

```diff
@@ -0,0 +1,179 @@
+import csv
+import sys
+from typing import List
+
+from weather_record import WeatherRecord
+from custom_types import (AverageTempHumidReportType,
+                          HighLowTempHumidityReportType)
+
+
+class WeatherDataAnalyzer:
+    """
+    Receives the file paths and store their record in the list. Then perform
+    different calculations on that list
+    """
+
+    def __init__(self, filepaths) -> None:
+
+        self.file_records: List[WeatherRecord] = []
+
+        self.file_data_reader_and_add_to_file_records(filepaths)
+
+    def file_data_reader_and_add_to_file_records(self, filepaths):
+        """
+        Read the data of all files given in filepaths and append it to the list
+
+        :param filepaths:
+        """
+
+        for filepath in filepaths:
+            with open(filepath, 'r') as file:
+                records_list = csv.DictReader(file)
+
+                for row in records_list:
+                    record = WeatherRecord(row)
+                    self.file_records.append(record)
+
+    def calc_low_and_high_temp_and_humid(self):
+
+        if len(self.file_records) == 0:
+            return None
+
+        return self.get_low_high_temp_and_max_humidity()
+
+    def get_low_high_temp_and_max_humidity(self):
+        """
+        From the file_records list, it returns the max of high temperature, min
+        of low temperature and max of high humidity
+
+        :return: HighLowTempHumidityReportType
+        """
+
+        valid_high_temp_records = self.remove_none_values_of_max_temp()
+        valid_low_temp_records = self.remove_none_values_of_min_temp()
+        valid_high_humidity_records = self.remove_none_values_of_max_humidity()
+
+        max_high_temp = max(
+            valid_high_temp_records, key=lambda x: int(x.max_temp)
+        )
+        min_low_temp = min(
+            valid_low_temp_records, key=lambda x: int(x.min_temp)
+        )
+        max_high_humidity = max(
+            valid_high_humidity_records, key=lambda x: int(x.max_humidity)
+        )
+
+        return HighLowTempHumidityReportType(
+            highest_temperature=max_high_temp,
+            lowest_temperature=min_low_temp,
+            high_humidity=max_high_humidity
+        )
+
+    def remove_none_values_of_max_temp(self):
+        """
+        Removes the records from file_records whose max_temperature value is None
+
+        :return: List[WeatherRecords]
+        """
+
+        valid_max_temps = [
+            rec for rec in self.file_records if rec.max_temp
```

Comment:
an alternate way to remove None can be like this

`return list(filter(lambda rec: rec.max_temp is not None, self. file_records))`